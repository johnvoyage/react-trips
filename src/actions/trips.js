import uuid from 'uuid';
import database from '../firebase/firebase';

export const startAddTrip = (tripData = {}) => {
  return (dispatch) => {
    const {
      country = ''
    } = tripData;
    const trip = {
      country
    };

    return database.ref('trips').push(trip).then((ref) => {
      dispatch(addTrip({
        id: ref.key,
        ...trip
      }));
    });
  };
};

export const addTrip = (trip) => ({
  type: 'ADD_TRIP',
  trip
});

export const removeTrip = ({ id } = {}) => ({
  type: 'REMOVE_TRIP',
  id
});

export const editTrip = (id, updates) => ({
  type: 'EDIT_TRIP',
  id,
  updates
});
