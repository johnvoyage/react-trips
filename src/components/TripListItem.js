import React from 'react';
import { Link } from 'react-router-dom';


const TripListItem = ({ dispatch, id, country }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{ country }</h3>
    </Link>

  </div>
);

export default TripListItem;