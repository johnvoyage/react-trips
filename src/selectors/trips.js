export const getVisibleTrips = (trips, { text, sortBy, startDate, endDate }) => {
  return trips.filter((trip) => {
    const startDateMatch = typeof startDate !== 'number' || trip.createdAt >= startDate;
    const endDateMatch = typeof endDate !== 'number' || trip.createdAt <= endDate;
    const textMatch = trip.country.toLowerCase().includes(text.toLowerCase());

    return startDateMatch && endDateMatch && textMatch;
  }).sort((a, b) => {
    if (sortBy === 'date') {
      return a.createdAt < b.createdAt ? 1 : -1;
    } else {
      return 1;
    }
    //else if (sortBy === 'amount') {
    //   return a.amount < b.amount ? 1 : -1;
    // }
  });
};

