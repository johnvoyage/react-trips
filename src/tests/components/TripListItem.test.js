import React from 'react';
import { shallow } from 'enzyme';
import trips from '../fixtures/trips';
import TripListItem from '../../components/TripListItem';

test('should render TripListItem correctly', () => {
  const wrapper = shallow(<TripListItem {...trips[0]} />);
  expect(wrapper).toMatchSnapshot();
});
