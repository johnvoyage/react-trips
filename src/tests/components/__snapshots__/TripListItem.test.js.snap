// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TripListItem correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/123abc"
  >
    <h3>
      Germany
    </h3>
  </Link>
</div>
`;
